src/molecules/BottomSheet/BottomSheet.native.js

```diff
@@ -0,0 +1,125 @@
+
+const styles = {
+  overlayStyle: ({ theme }) => {
+    return {
+      backgroundColor: theme.colors.shade[950],
+    };
+  },
+  childrenStyle: ({ theme }) => {
+    return {
+      backgroundColor: theme.colors.background[200],
+    };
+  },
+};
+
+const HeaderContainer = styled(View)`
+  background-color: ${(props) => props.theme.colors.background[200]};
+  border-top-right-radius: 8px;
+  border-top-left-radius: 8px;
+  box-shadow: ${(props) => `0px -4px 15px ${props.theme.colors.primary[930]}`};
+`;
+
+const BottomSheetDragBar = styled(View)`
+  background-color: ${(props) => props.theme.colors.shade[920]};
+  border-radius: 4px;
+`;
+
+const BottomSheet = forwardRef(
+  (
+    {
+      snapPoint = DEFAULT_SNAP_POINT,
+      children,
+      HeaderComponent,
+      FooterComponent,
+      onBackDropPress = () => {},
+      onBackButtonPress = () => {},
+      onOpened = () => {},
+      onClosed = () => {},
+      childrenStyle = {},
+      adjustToContentHeight = false,
+      overlayStyle = {},
+      alwaysOpen = 0,
+    },
+    ref,
+  ) => {
+    const theme = useTheme();
+
+    return (
+      <RNModalize
+        ref={ref}
+        snapPoint={snapPoint}
+        HeaderComponent={
+          <HeaderContainer>
+            <Flex alignItems="center">
+              <Space padding={[1, 0, 1.5, 0]}>
+                <View>
+                  <Size height={0.5} width={8}>
+                    <BottomSheetDragBar />
+                  </Size>
+                </View>
+              </Space>
+            </Flex>
+            {HeaderComponent}
+            <Divider color="shade.920" horizontal />
+          </HeaderContainer>
+        }
+        FloatingComponent={
+          FooterComponent ? (
+            <Position position="absolute" bottom={0} left={0} right={0}>
+              <View>
+                <Divider color="shade.920" horizontal />
+                {FooterComponent}
+              </View>
+            </Position>
+          ) : null
+        }
+        overlayStyle={{ ...styles.overlayStyle({ theme }), ...overlayStyle }}
+        onOverlayPress={onBackDropPress}
+        avoidKeyboardLikeIOS={true}
+        onOpened={onOpened}
+        onClosed={onClosed}
+        onBackButtonPress={onBackButtonPress}
+        childrenStyle={{ ...styles.childrenStyle({ theme }), ...childrenStyle }}
+        withHandle={false}
+        panGestureComponentEnabled={true}
+        adjustToContentHeight={adjustToContentHeight}
+        alwaysOpen={alwaysOpen}
+      >
+        {children}
+      </RNModalize>
+    );
+  },
+);
+
+BottomSheet.displayName = 'BladeBottomSheet';
+
+BottomSheet.propTypes = {
+  snapPoint: PropTypes.number,
+  children: PropTypes.node,
+  HeaderComponent: PropTypes.node,
+  FooterComponent: PropTypes.node,
+  onOpened: PropTypes.func,
+  onClosed: PropTypes.func,
+  onBackButtonPress: PropTypes.func,
+  onBackDropPress: PropTypes.func,
+  childrenStyle: PropTypes.object,
```

Comment:
children style will be defined by DS and not consumer, isn't it?

Reply:
I too had this confusion. Default styles should be from DS.